tensorboard/plugins/interactive_inference/witwidget/notebook/visualization.py

```diff
@@ -441,3 +450,80 @@ def set_compare_custom_predict_fn(self, predict_fn):
     if not self.has_compare_model_name():
       self.set_compare_model_name('2')
     return self
+
+  def _convert_json_to_tf_examples(self, examples):
+    tf_examples = []
+    for json_ex in examples:
+      ex = tf.train.Example()
+      for feat in json_ex:
+        if isinstance(json_ex[feat], (int, long)):
+          ex.features.feature[feat].int64_list.value.append(json_ex[feat])
+        elif isinstance(json_ex[feat], Number):
+          ex.features.feature[feat].float_list.value.append(json_ex[feat])
+        else:
+          ex.features.feature[feat].bytes_list.value.append(
+            json_ex[feat].encode('utf-8'))
+      tf_examples.append(ex)
+    return tf_examples
+
+  def set_cmle_model(self, project, model, version=None):
+    """Sets the model information for a model served by CMLE.
+
+    CMLE is Google's Cloud Machine Learning Engine.
+
+    Args:
+      project: The name of the CMLE project.
+      model: The name of the CMLE model.
+      version: Optional, the version of the CMLE model.
+
+    Returns:
+      self, in order to enabled method chaining.
+    """
+    self.set_inference_address(project)
+    self.set_model_name(model)
+    self.store('use_cmle', True)
+    if version is not None:
+      self.set_model_signature(version)
+    return self
+
+  def set_compare_cmle_model(self, project, model, version=None):
+    """Sets the model information for a second model served by CMLE.
+
+    CMLE is Google's Cloud Machine Learning Engine.
+
+    Args:
+      project: The name of the CMLE project.
+      model: The name of the CMLE model.
+      version: Optional, the version of the CMLE model.
+
+    Returns:
+      self, in order to enabled method chaining.
+    """
+    self.set_compare_inference_address(project)
+    self.set_compare_model_name(model)
+    self.store('use_cmle_compare', True)
+    if version is not None:
+      self.set_compare_model_signature(version)
+    return self
+
+  def set_target_feature(self, target):
+    """Sets the name of the target feature in the provided examples.
+
+    If the provided examples contain a feature that represents the target
+    that the model is trying to predict, it can be specified by this method.
+    This is necessary for CMLE models so that the target feature isn't sent
+    to the model for prediction, which can cause model inference errors.
+
+    Args:
+      target: The name of the feature in the examples that represents the value
+      that the model is trying to predict.
+
+    Returns:
+      self, in order to enabled method chaining.
+    """
+    self.store('target_feature', target)
+    return self
+
+  def _set_uses_json_input(self, is_json):
```

Comment:
This may need edits after force_json_conversion flag as the main model and compare model may have different inputs in that case.

Reply:
This still works, as in the cmle infer method we convert from tf.example to json if their uses_json_input is true (meaning the examples were initially provided as JSON) ,or if for that model the new override argument was set to true (meaning the examples were provided as tf.Example but for this model we want to convert to json to send it to the model).